sys-utils/lscpu-virt.c

```diff
@@ -454,19 +455,23 @@ void vmware_bdoor(uint32_t *eax, uint32_t *ebx, uint32_t *ecx, uint32_t *edx)
 }
 
 static jmp_buf segv_handler_env;
+static sigset_t oset;
 
 static void
 segv_handler(__attribute__((__unused__)) int sig,
              __attribute__((__unused__)) siginfo_t *info,
              __attribute__((__unused__)) void *ignored)
 {
+	if (sigprocmask(SIG_SETMASK, &oset, NULL))
+		err(EXIT_FAILURE, _("cannot set signal mask"));
```

Comment:
Could this be moved out of the signal handler context?
(I'm not strong with signals, so if that doesn't make sense ignore it)

Reply:
I think it could be moved to `sigsetjmp` branch. Make sure that the logic of the signal handler function is simple.